src/configure-device.js

```diff
@@ -291,24 +368,44 @@ module.exports = function (RED) {
         }
     };
 
-    RED.httpAdmin.get('/node-red-teknoir-configure-devices', function (req, res) {
-        let client = new Client({version: '1.13'});
-        let data = fs.readFileSync(require.resolve('./kubeflow.org_devices.yaml'), {encoding: 'utf8', flag: 'r'});
-        let crd = yaml.load(data)
-        client.addCustomResourceDefinition(crd);
-        client.apis['kubeflow.org'].v1.namespaces(namespace).devices().get()
-            .catch(error => res.status(500).send(error))
-            .then(devices => {
-                deviceList = [];
-                devices.body.items.forEach((device) => {
-                    deviceList.push({
-                        name: device.metadata.name,
-                        namespace: device.metadata.namespace,
-                        labels: device.metadata.labels
-                    });
-                })
-                res.status(200).json(deviceList);
-            });
+
+
+    RED.httpAdmin.get('/node-red-teknoir-configure-devices', async function (req, res) {
+        try {
+            let client = new Client({version: '1.13'});
+
+            let kubeflowData = fs.readFileSync(require.resolve('./kubeflow.org_devices.yaml'), {encoding: 'utf8', flag: 'r'});
+            let kubeflowCrd = yaml.load(kubeflowData);
+            client.addCustomResourceDefinition(kubeflowCrd);
+
+            let teknoirData = fs.readFileSync(require.resolve('./teknoir.org_devices.yaml'), {encoding: 'utf8', flag: 'r'});
+            let teknoirCrd = yaml.load(teknoirData);
+            client.addCustomResourceDefinition(teknoirCrd);
+
+            let kubeflowDevices = await client.apis['kubeflow.org'].v1.namespaces(namespace).devices().get();
+            let kubeflowDeviceList = kubeflowDevices.body.items.map(device => ({
```

Comment:
Could this fail if we remove the CRD? Does it need a try catch block?

Reply:
you're right! fixed